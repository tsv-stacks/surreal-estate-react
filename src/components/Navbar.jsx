import React from "react";
import "../styles/Navbar.css";
import { ReactComponent as Castle } from "../assets/castle-logo.svg";

export default function Navbar() {
  return (
    <nav className="navbar">
      <div className="nav-container">
        <div className="logo-icon flex">
          <Castle color="purple" className="navbar-logo" />
          <h2 className="navbar-logo-text">Surreal Estate</h2>
        </div>
        <h2 className="navbar-links-item">View Properties</h2>
        <h2 className="navbar-links-item">Add a Property</h2>
      </div>
    </nav>
  );
}
