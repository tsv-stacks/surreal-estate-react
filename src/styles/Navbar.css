.nav-container {
  background-color: #f5f5f5;
  height: 80px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 40px;
  width: 55%;
}

.navbar {
  background-color: #f5f5f5;
  height: 80px;
}

.logo-icon {
  display: flex;
  align-items: center;
}

.navbar-links-item {
  margin: 0;
  padding: 10px;
  font-size: 20px;
  font-weight: 500;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
}

.navbar-links-item:hover {
  background-color: purple;
  color: white;
}
.navbar-logo-text {
  font-weight: 700;
  margin: 0;
}

@media (max-width: 768px) {
  .nav-container {
    width: 100%;
  }
}

@media (max-width: 425px) {
  .nav-container {
    height: 60px;
    padding: 0 10px;
    text-align: center;
  }
  .navbar {
    height: 60px;
  }
  .navbar-links-item {
    font-size: 16px;
  }
  .navbar-logo-text {
    font-size: 14px;
    text-align: left;
  }
}

@media (max-width: 325px) {
  .nav-container {
    height: 55px;
    padding: 0 5px;
    text-align: center;
  }
  .navbar {
    height: 55px;
  }
  .navbar-links-item {
    font-size: 14px;
  }
  .navbar-logo-text {
    font-size: 14px;
    text-align: left;
  }
}
