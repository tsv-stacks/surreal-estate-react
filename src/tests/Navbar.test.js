import React from "react";
import { render } from "@testing-library/react";
import Navbar from "../components/Navbar";

describe("Navbar Component", () => {
  test("Navbar is rendered correctly", () => {
    const { getByText } = render(<Navbar />);

    expect(getByText(/View Properties/i)).toBeInTheDocument();
    expect(getByText(/Add a Property/i)).toBeInTheDocument();
  });

  test("Navbar is rendered correctly - snapshot", () => {
    const { getByText } = render(<Navbar />);

    expect({ getByText }).toMatchSnapshot();
  });
});
