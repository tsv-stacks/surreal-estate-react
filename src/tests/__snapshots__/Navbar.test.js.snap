// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Navbar is rendered correctly - snapshot 1`] = `
Object {
  "getByText": [Function],
}
`;
